File: backend/app/api/v1/schemas/file.py
# FileResponse, PresignedUrlResponse

from pydantic import BaseModel
from datetime import datetime


class FileResponse(BaseModel):
    id: str
    request_id: str
    original_name: str
    mime_type: str
    size_bytes: int
    page_count: int
    object_key: str
    created_at: datetime

    class Config:
        from_attributes = True


class PresignedUrlResponse(BaseModel):
    url: str
    expires_at: datetime

File: 01_frontend/src/types/file.ts
// File types

export interface FileInfo {
  id: string
  request_id: string
  original_name: string
  mime_type: string
  size_bytes: number
  page_count: number
  object_key: string
  created_at: string
}

export interface UploadFile {
  file: File
  id: string
  progress: number
  status: 'pending' | 'uploading' | 'completed' | 'error'
  error?: string
}

export interface UploadConfig {
  method: string
  tier: number
  output_format: 'txt' | 'json' | 'md'
  retention_hours: number
}

export interface PresignedUrlResponse {
  url: string
  expires_at: string
}
